### DISCUSSION TITLE
Bouton déconnexion

### DISCUSSION THREAD

**Discussion by ABIAdrienG** [witsa/synapps#588] [Q&A] [answered]
Bonjour,

Je travail actuellement sur un bouton de déconnexion permettant le renvoi à la page de base et non à la page Synapps de connexion.

Actuellement le bouton utilise un onclick "context.dataStores.redy.logout();"
Le problème rencontré avec cette méthode est que en réalité cette fonction n'est pas une déconnexion mais une redirection de page.

J'aimerai utiliser la fonction interne ?DoLogOff située apparemment dans le "/easy/GUI/Menu/MM2/MS3".
![image](https://github.com/witsa/synapps/assets/127521459/f12fa758-b2cd-4952-86f7-fb4e5c5cb01c)

La fonction ?DoLogOff semble **terminée le temps de session du SID** et cela correspond parfaitement à ma demande.

Est-ce que ça serait possible de refaire exactement cet fonction sur la couche HTML synapps ?
Si oui, pouvez-vous m'indiquer une solution svp ?

je vous remercie par avance.

**Comment by manufitoussiwit** [comment, marked as answer]
Bonjour,

Il est possible de faire ça en remplaçant le code du click du bouton par : 

```js
if (context.dataStores.redy.session.inSid) {
  const origin = context.dataStores.redy.host.domain;
  const sid = context.dataStores.redy.session.inSid;
  const url = `${origin}/WSID${sid}/?DoLogOff`;
  context.dataStores.redy.disconnect();
  document.location = url;
} else {
  context.dataStores.redy.logout();
}

```
ça va déconnecter et renvoyer sur la page de login par défaut du REDY.
Si la synapp est affichée indépendamment du REDY, ça affichera le login de synapp uniquement.

++
